AKS Engine CLI guide (#533)
microsoft/azure_arc: docs/azure_arc_jumpstart/azure_arc_k8s/aksengine/aksengine_azcli/_index.md

---
type: docs
title: "AKS Engine Kubernetes cluster in Azure with Azure CLI"

> linkTitle: "AKS Engine Kubernetes cluster in Azure with Azure CLI"

Reviewer: No reference to Arc. Please update based on other scenarios

weight: 1
description: >
---

> ## AKS Engine in Azure 

Reviewer: See previous comment

## Installing AKS Engine

* AKS Engine requires a binary installed in the system. You can refer to [AKS Engine Releases](https://github.com/Azure/aks-engine/releases/latest) to get the latest version, the following example downloads and installs version 0.61.0:

```bash
# Install AKS engine
# Go here: https://github.com/Azure/aks-engine/releases/latest
# Example with v0.61.0:
aksengine_exec=$(which aks-engine)
if [[ -n "$aksengine_exec" ]]
then
    echo "aks-engine executable found in ${aksengine_exec}"
else
    echo "Downloading and installing aks-engine executable..."
    aksengine_tmp=/tmp/aksengine.tar.gz
    wget https://github.com/Azure/aks-engine/releases/download/v0.61.0/aks-engine-v0.61.0-linux-amd64.tar.gz -O $aksengine_tmp
    tar xfvz $aksengine_tmp -C /tmp/
    sudo cp /tmp/aks-engine-v0.61.0-linux-amd64/aks-engine /usr/local/bin
fi

> ```

Reviewer: This section should be embedded in the main deployment script

> 

Reviewer: Please add Automation Flow section

## Deployment

> In this guide we will generate an AKS Engine cluster with 1 master node and 2 worker nodes. There are additional examples for the required JSON files with different configurations in [AKS Engine Examples](https://github.com/Azure/aks-engine/tree/master/examples). 

Reviewer: ```suggestion
In this guide, we will generate an AKS Engine cluster with 1 control plane node and 2 worker nodes. There are additional examples for the required JSON files with different configurations in [AKS Engine Examples](https://github.com/Azure/aks-engine/tree/master/examples). 
```

> * Before initiating the deployment, edit the [*arc_k8s_aksengine.sh script*](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/azure_arc_k8s_jumpstart/aksengine/arc_k8s_aksengine.sh) script so that the environment variables match your setup:

Reviewer: ```suggestion
* Before initiating the deployment, edit the [*arc_k8s_aksengine.sh script*](https://github.com/erjosito/azure_arc/blob/aksengine/azure_arc_k8s_jumpstart/main/arc_k8s_aksengine.she.sh) script so that the environment variables match your setup:
```

Also, change the folder name from _aksengine_ to _aks-engine_.

>   * k8s_rg: Azure Resource Group where all the AKS Engine resources get deployed. For example: arc-engine-demo
>   * location: Azure location where the AKS Engine resource will be created in Azure. For example: westeurope
>   * arc_rg: Azure Resource Group to connect your Azure Arc enabled cluster. For example: arc-engine-demo
>   * arc_name: Name of for the Azure Arc connect cluster. For example: aks-engine-demo

Reviewer: ```suggestion
  * _k8s_rg_: Azure Resource Group where all the AKS Engine resources get deployed. For example, _arc-engine-demo_
  * _location_: Azure location where the AKS Engine resource will be created in Azure. For example, _westeurope_
  * _arc_rg_: Azure Resource Group to connect your Azure Arc enabled cluster. For example, _arc-engine-demo_
  * _arc_name_: Name of for the Azure Arc connected Kubernetes cluster. For example, _aks-engine-demo_
```

> * To start the automation, you will need to run the [arc_k8s_aksengine.sh script](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_k8s/azure_arc_k8s_jumpstart/aksengine/arc_k8s_aksengine.sh) to do that run the command below: 

Reviewer: ```suggestion
* To start the automation, you will need to run the [arc_k8s_aksengine.sh script](https://github.com/erjosito/azure_arc/blob/aksengine/azure_arc_k8s_jumpstart/aksengine/arc_k8s_aksengine.sh) to do that run the command below: 
```

Also, change the folder name from _aksengine_ to _aks-engine_.

```bash
./arc_k8s_aksengine.sh
```

> * The script will:
>   * Set up the environment variables that match your custom deployment.
>   * Create the *aksengine_cluster.json* file to configure AKS Engine in Azure.
>   * Use the *aks-engine* executable to create the cluster.
>   * Export the KUBECONFIG variable to be able to connect to the cluster.
>   * Onboard the cluster onto Azure Arc.

Reviewer: ```suggestion
* The script will:
  * Set up the environment variables that match your custom deployment.
  * Create the _aksengine_cluster.json_ file to configure AKS Engine in Azure.
  * Use the _aks-engine_ executable to create the cluster.
  * Export the KUBECONFIG variable to be able to connect to the cluster.
  * Onboard the cluster onto Azure Arc.
```
